assets/js/editor-components/compatibility-notices/use-compatibility-notice.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+import { useLocalStorageState } from '@woocommerce/base-hooks';
+
+const initialDismissedNotices: string[] = [];
+
+export const useCompatibilityNotice = (
+	blockName: string
+): [ boolean, () => void ] => {
+	const [ dismissedNotices, setDismissedNotices ] = useLocalStorageState(
+		`wc-blocks_dismissed_compatibility_notices`,
+		initialDismissedNotices
+	);
+	const [ isVisible, setIsVisible ] = useState( false );
+
+	const isDismissed = dismissedNotices.includes( blockName );
+	const dismissNotice = () => {
+		const dismissedNoticesSet = new Set( dismissedNotices );
```

Comment:
Sets are not supported in ie11. Not sure if we have to support it or our configuration of @babel/preset-env supports this, but I'd double-check it in the browser. I saw it's used once in the project outside of notices.

Reply:
Good catch. I think Babel takes care of transpilling this, we have a couple of other instances of Set in the codebase. I just tested with BrowserStack to double-check and everything seemed to work fine with IE11. :ok_hand: